### ISSUE TITLE
Set Trigger Type panel (STT) development

### ISSUE THREAD

**Issue by AkyV** [MontyTRC89/Tomb-Editor-Issues#13]
Features copied from NGLE STT:
1. "Find" different windows content
2. Open different windows content in a TXT.
3. Export AnimCommand (not Export Script, that already exists, I can see :) )
4. "Find Trigger Number" - to open STT at the required trigger: for trigger IDs (like F84), exported script triggers, exported AnimCommand triggers (and, in the future, exported triggers in the C++ plugin code as functions)
5. Trigger's Remark window for short descriptions.
6. Trigger's Help button (see eg. A41 in NGLE) for long descriptions.

**Comment by ActuallyaDeviloper** [comment]
A search function is certainly planned. :)

Can you elaborate on the purpose of the trigger remark and trigger help button? I opened action 41 in NGLE but pressing the help button does not appear to do anything.

btw, for us it is much easier to keep track of open concerns if you create one issue per concern. The requested features will most likely not arrive simultaneously and then we can't really close it even though most stuff is now done. :)

**Comment by AkyV** [comment]
- Oh, first I misunderstood that issue #3. I thought that is about search in the map... (Speaking of, are you planning to enhance searching tools in the map? See eg. Advanced Search of NGLE.)
- By any chance are you using Win 8.1 for NGLE? I experienced the same bug in STT with NGLE. I must say that the perfect OS for NGLE seems XP. I have an old XP PC almost only to use NGLE. :)
I suggest installing NGLE on an OP like that, for further attempts. :)
Btw here is the documentation for remark/help:

http://www.trlevelmanager.eu/plugin_sdk_help/triggers.htm#LINK52

- One concern per one issue, even for tiny issues. I understood. :)

**Comment by ActuallyaDeviloper** [comment]
You are right, Monty does indeed not write very elaborate issue reports. :D
However it mentions the search being inside the trigger window, so I read this is the feature requested here.

What are you missing in the existing search window? (Edit -> Search or Ctrl+F)
I was personally responsible for implementing it and it was really important to me that the new search dialog is more convinient to use. For this reason it operated in realtime and is using fuzzy search, that means there is a good chance it will find what you are looking for even if you are making a spelling mistake. 
As far as I am aware, it is a superset of what NGLE was able to do.

Yes, I am in fact using Windows 8.1.
Thanks for linking the document. I searched in the data files available to us for REMARK and I found nothing. Maybe someone for whom that feature of ngle works can go manually through all triggers and make a list with all the remarks? We could feed that then into our ng trigger information file.
Otherweise I guess MontyTRC would have to contact Paolone again in the hope that will share those files.

**Comment by AkyV** [comment]
Search:
Your window has: Everything/Room/AllObjects/ObjectsinCurrentRoom/ItemType.
NGLE has: Object(WADObject)/Indices of all Item/Floor Type/Trigger(Activation mode)/Trigger (What-to)/Trigger Plugin/Trigger not handled/Effect (light etc.)/Texture - also having sub-menus for them.
Plus, what about old TRLE search, like Trigger->Object, Illegal slopes etc.?

Remark/Help:
A41 was only an example from Paolone.  Classic trng.dll triggers have not been filled yet with data like that. (If you give me a proper form, then I can send the those data for trng.dll triggers. But it will last months, probably I can send a few triggers per week, if you want.)
Plus, based by the document, plugin makers can also fill they own triggers for that data.

**Comment by ActuallyaDeviloper** [comment]
You can put any keywords in the search dialog, including for the things you have listed.
There is a special view mode for illegal slope (the exclamation mark in the toolbar) and for triggers you can open them and go to them.

Ah, so this is only relevant for plugins?
Currently we haven't put thought into plugins yet, basically that's not supported currently.

**Comment by AkyV** [comment]
Search:
Now it is clear. Thanks! :) (An advice: video tutorial/readme or not, you should also release a PDF manual after each non-beta/test release. Perhaps a basic one, and later only with the updates.)

Remark/Help:
No.:)
The feature has two parts:
- Plugin makers can add their remarks/helps to the plugin trigger.
- Paolone can add his remarks/helps to the original TRNG (non-plugin) trigger.
So let'see only the "Paolone part".  He showed A41 only as an example. He had no time to fill them all the original TRNG triggers with data like that, since it is a brand new feature. That is why I said, as a "TRNG expert" (I can say... :) ) that I will give you data about that, but it take time. :)

**Comment by ActuallyaDeviloper** [comment]
Yes we are very poor with documentation and stuff. :(
Unfortunately writing documentation and whatnot feels like a tedious task to us programmers.
We appreaciate any help on this front!

**Comment by AkyV** [comment]
PDF:
Just now I can't really help myself, being still very amateur to Tomb Editor. Later. :)
However, perhaps you could ask people on the forums to also help with that. :)

Remark/Help:
- About plugin part: I release my first plugin soon, with a tons of remark/help like that,  What I want to say with it is perhaps other plugin makers will also do that, so at the latest when you implement plugin tools, you should add this feature. :) If you do everything well, then installing a plugin will fill that data for its own plugin automatically.
- Paolone's part: I can promise I will send this data to you, even if Paolone doesn't write it, probably in a random day of 2018,  It is not a plugin, so I don't know how to upload that, so I will send it a "Trigger code/Remark's contents/Help's contents" form, and hopefully you will add it somehow.
